### ISSUE TITLE
Link search results to single page

### ISSUE THREAD

**Issue by roopeux** [buda-base/public-digital-library#967]
Based on [#964](https://github.com/buda-base/public-digital-library/issues/964)

Let's link search results to one page only.

The general highlight style changes from bold to red.
Because we won't have links within links, the whole box should be a link.
![Image](https://github.com/user-attachments/assets/d45da8e1-1faa-4a22-8c9a-ce89bd78af77)

**Highlights**
- If the match is in prefLabel, highlight the match in the main title.
- If the match is in altLabel, highlight the match in red in "Other titles".  Do not swap the altTitle as the main title.
- If the match is in any other metadata field, show and highlight that field similarily.

**Etext match only**
- If the match is only in the etext, link to the first occurrence in the etext.
- Mention the number of etext hits
![Image](https://github.com/user-attachments/assets/f6bd7582-4bf4-490b-9e2f-cc1d7d0b877b)

**Match in both etext and metadata (including titles)**
- Highlight the hit only in the metadata
- Show one snippet without highlight
- Show the etext hit count as above.

**Comment by eroux** [comment]
I like some of this, thanks!

The first thing that strikes me is that even though I'm only a little color blind, I really struggle to see the difference between the red and the black. I think there are tools to measure contrast right? What do they say? I know you don't like highlight, but I don't understand why you want to get rid of it so desperately, does it come from the user tests?

I think we should replace "Etext" by "Match in etext", what do you think?

I still really don't like that some results will lead to one type of page (etext viewer if results only in etext) and some to another (entity page when results in the metadata). One way to solve that would be something like a button "see all matches in etext" next to the  that would lead to search results in the etext viewer for that particular text... what do you think?

**Comment by roopeux** [comment]
Oh yes, that's a good point about color.  We should replace the red by background highlight.
The reason I did not like it was that red font on light yellow highlight would not pass the contrast test for accessibility, but black on light yellow will do.

**Comment by roopeux** [comment]
Regarding links to etext viewer, my thought is that we should take the user where the match is, nowhere else.  There is no consistency that users would expect in this regard.  The only thing they want to see is the match in some context.

Throwing them to the etext viewer might have been a problem when orientation was still weak, but with breadcrumbs it should not be a problem.  They will easily get to main entity page if they want.

I think it is a good idea to talk about this face-to-face to make sure we completely understand each other's points.

**Comment by roopeux** [comment]
Okay with "Match in etext"

**Comment by eroux** [comment]
sure, but my point is that there is no (or well, very little if you count the highlight) visual distinction between:
- match in etext + metadata
- match in etext only

If we want to have them be completely separate types of results we should introduce a proper visual distinction. Maybe in the case of match in etext only:
- instead of having the thumbnail the result could have an etext icon
- the match in etext could be first instead of the title, and then there would be "in:", then lower the title of the text

do you see what I mean?

**Comment by roopeux** [comment]
This clarifies a lot.  I'll design a new result for etext matches.  I agree that more visual distinction would be useful.

**Comment by eroux** [comment]
after our discussion the idea is to go in the direction of having two distincts results in the case where there is a match in the metadata and another in the etext. I think it's perfectly fine, although in searches where this case appears often it might be a bit strange, you would have systematically two consecutive results per text, which might be a bit bizarre... Also that probably means reworking the API a bit

**Comment by eroux** [comment]
> Continuing this thought, when user's keywords match the pref or alt label, we can assume that he wanted to match the title, and matches in etext are irrelevant. For example, many versions of "bka' 'gyur" will have "bka' 'gyur" in etext, but that is irrelevant and the etext results should not be shown.

As long as we show "Nb of hits in etext: ..." for these results we can try that yes

**Comment by roopeux** [comment]
@berger-n, could you check if the etext linking idea on this video is doable?  Some little questions for @eroux too.  (Turn sound on.)

https://github.com/user-attachments/assets/5a624b24-54fa-4804-81f6-a8f624890816

**Comment by eroux** [comment]
I think the link would be very good yes, I think it can work more or less diretly @berger-n ?

As you anticipated, I think the two types of results should be more visually distinct

**Comment by berger-n** [comment]
good idea indeed to open the etext inner search results, already thought about doing this actually, shouldn't be too complicated

**Comment by berger-n** [comment]
promising though not 100% complete: [link](https://library-dev.bdrc.io/osearch/search?etext_access%5B0%5D=3&q=spyod%20%27jug%20gi%20%27grel%20pa&page=2)

![simplescreenrecorder-2024-11-26_18 36 18 mkv](https://github.com/user-attachments/assets/c550611f-c85a-4cee-b683-81d2c2022012)

(working on the case when there are multiple matches in the same volume and we want to open a specific one)

**Comment by roopeux** [comment]
@eroux I edited the original issue and changed the priority of matches to 1 labels, 2 etext, 3 other metadata.  We don't have a separate case for 2 + 3 which was discussed above, because it happens very rarely (checked with batch search).  We don't have cases for any other pairs except 1 + 3.

**Comment by berger-n** [comment]
> working on the case when there are multiple matches in the same volume and we want to open a specific one

now works in this case as well: https://library-dev.bdrc.io/osearch/search?author%5B0%5D=P6962&q=spyod%20%27jug

![simplescreenrecorder-2024-11-27_18 17 56 mkv](https://github.com/user-attachments/assets/d89e455e-1a87-4937-abab-311b5b1b93bf)

**Comment by roopeux** [comment]
Great job @berger-n !

**Comment by roopeux** [comment]
The design is ready for my part.  I did not anticipate confusion between the entity and etext results even before but now that the "Etext" title is highlighted in red, "Etext" repeated three times in the top section, and the etext snippet is on top with a clear highlight, the distinction should be clear enough.

**Comment by eroux** [comment]
ok, once @berger-n let's review it and have Jann test it

**Comment by berger-n** [comment]
done: https://library-dev.bdrc.io/osearch/search?q=dri%20med%20%27od%20zer&etext_access%5B0%5D=3

![image](https://github.com/user-attachments/assets/0fca3b3d-0b54-405b-9589-73608a88d5cf)
---

![image](https://github.com/user-attachments/assets/3710f349-a70d-4fa2-ac67-dc4cd961afe8)
---

oh just found a glitch on that one, when switching on/off the exclude commentaries filter:

![image](https://github.com/user-attachments/assets/a4aade88-62fc-4c14-8a01-b8c33d62c383)

**Comment by eroux** [comment]
Thanks! I think that's pretty good!

@berger-n One small fix: on the etext results in https://library-dev.bdrc.io/osearch/search?q=dag%20yig%20rgyas%20pa the link works everywhere except on the etext match itself (on this image, the part in blue should be the same link as the rest but it's not a link, nothing happens when I click on it)

![Capture d’écran de 2024-12-06 21-46-28](https://github.com/user-attachments/assets/fb52100c-1d11-44c9-a527-b6584ceb90ae)

@roopeux I still think the two types of results should be more differenciated. Why not something like:
- instead of just the plain thumbnail, I propose we have something distinctive so that we see clearly that they are different types of results (because this part of the result is really what represents the result type, for instance the person icon, place icon, etc.). We can keep the thumbnail but add maybe the etext icon in a circle at the bottom of the thumbnail (inside it), would that be ok?
- in the result itself, instead of "Etext: ", we could say maybe "Match: "
- instead of "hits in etext: " we could have "Nb. matches: "
- instead of the title of the text we could maybe have "Etext for: [title]"

what do you think?

**Comment by roopeux** [comment]
<img width="695" alt="image" src="https://github.com/user-attachments/assets/55abacbd-cdd5-48fd-a17e-a51cdb4aa426">

There is not really such thing as an etext icon, but we can display it as a tag, overlaid on the thumbnail.
"Etext for:" in italics
The snippet will not need any label, because it follows the standard everyone understands.
It is still good to say "Matches in etext" to make it clear.

**Comment by eroux** [comment]
Looks good to me, thanks!

**Comment by roopeux** [comment]
@berger-n , the last layout should now be final.  [Figma.](https://www.figma.com/design/jagzws5rTk1VVPfwa62XBy/Figma-basics?node-id=3086-3637&node-type=frame&t=znEuKvfp04Z9WLyl-11)

**Comment by eroux** [comment]
BTW, the etext icon I was talking about is

![WORKSEARCH_Etext_petit](https://github.com/user-attachments/assets/59efadbf-a3ab-45b3-abdd-ea29b627c14e)

see

![Capture d’écran de 2024-12-07 09-14-24](https://github.com/user-attachments/assets/1c0b5405-303d-4cde-9c05-055d6fdb49ba)

**Comment by eroux** [comment]
(edited the comment a bit) @roopeux maybe we could use that icon next to "Etext" (or just replace it?)

**Comment by roopeux** [comment]
We need to have either icon and label or label.  Icon only will not work, because if you present that icon to 1000 people, only a small percentage will say it means etext.

**Comment by eroux** [comment]
yes, and also if you ask 1000 people what etext means only a small percentage will have the right answer

**Comment by berger-n** [comment]
fixed/done: https://library-dev.bdrc.io/osearch/search?q=dag%20yig%20rgyas%20pa

![image](https://github.com/user-attachments/assets/e8d4b1f2-4fec-4d5c-a46e-07e67b1789c3)
